### PR TITLE
Proposing usage example in README

### PR DESCRIPTION
**Pull request by fiddur**
Adding a simple usage example, based on http://oiccli.readthedocs.io/en/latest/intro.html#a-conversation but simplified and fitted to javascript.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+OpenID Connect Relying Party
```
Comment (TMSCH) [inline README.md]: OpenID Connect Client?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Getting an `identifier` from the user, using webfinger to discover the OP
+issuer, getting openid-configuration and using that to register a new client.
+
+```javascript
+const oidcclient = require('oidc-client');
+
+/**
+ * Get a client to use, either from storage or register a new one.
+ */
+async function getClient({ identifier, getClient, registerClient, clientMetadata }) {
```
Comment (TMSCH) [inline README.md]: Remove space between curly brackets and arguments.

Comment (TMSCH) [inline README.md]: Could we use a different name for this function? As one of the destructured argument is also named getClient, it becomes confusing.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  // Check existing clien storage first.
+  const oldClient = await getClient(issuer);
+  if (oldClient) return oldClient;
+
+  // Prepare dynamic registration.
+  const openidConfig = await oidcclient.getOIDCConfig(issuer);
+
+  // Register a new client.
+  const client = await oidcclient.register({ openidConfig, clientMetadata });
```
Comment (TMSCH) [inline README.md]: Same here for space after and before curly brackets.

**Review by TMSCH — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (fiddur) [reply README.md]: Yep, sloppy copy :P

File: README.md (hunk 2 above)
Comment (fiddur) [reply README.md]: Good catch.  Actually, eslint would have caught this and the other things, but it's not setup to check examples in a markdown file.  But I'll need to conclude the discussions with Roland about what belongs where first, then I'll update it.

**Pull request closed without merging**